### PR TITLE
Ensure model classes exists in configuration

### PR DESCRIPTION
**Pull request by pyrech**
Ref #818

### CODE REVIEW

File: DependencyInjection/Configuration.php [hunk 1]
```diff
@@ -388,9 +388,33 @@ private function addModelSection(ArrayNodeDefinition $node)
                 ->arrayNode('class')
                     ->addDefaultsIfNotSet()
                     ->children()
-                        ->scalarNode('media')->defaultValue('Application\\Sonata\\MediaBundle\\Entity\\Media')->end()
-                        ->scalarNode('gallery')->defaultValue('Application\\Sonata\\MediaBundle\\Entity\\Gallery')->end()
-                        ->scalarNode('gallery_has_media')->defaultValue('Application\\Sonata\\MediaBundle\\Entity\\GalleryHasMedia')->end()
+                        ->scalarNode('media')
+                            ->defaultValue('Application\\Sonata\\MediaBundle\\Entity\\Media')
+                            ->validate()
+                                ->ifTrue(function ($v) {
```
Comment (Soullivaneuh) [inline DependencyInjection/Configuration.php]: Why not just `ifFalse` instead of a reversed boolean result?

**Comment by Soullivaneuh** [comment]
Not sure if configuration is the proper place for testing those cases.

Would not be better on the configuration processing?

Also, could you add tests about it?

Feel free to use [matthiasnoback/symfony-dependency-injection-test](https://packagist.org/packages/matthiasnoback/symfony-dependency-injection-test) package if this can help in your case.

File: DependencyInjection/Configuration.php (hunk 1 above)
Comment (pyrech) [inline DependencyInjection/Configuration.php]: Are you sure the `ifFalse()` method exists? Can't see it in [ExprBuilder](https://github.com/symfony/symfony/blob/2.8/src/Symfony/Component/Config/Definition/Builder/ExprBuilder.php).

Comment (Soullivaneuh) [inline DependencyInjection/Configuration.php]: Indeed. Making sense to me having a `ifFalse`.

Ignore my comment so. :+1:

**Comment by pyrech** [comment]
IMO, the validation should be done inside the Configuration class, not the extension (which is more related to merge and build a clean config).

I will have a look to add tests for this ;)

**Comment by Soullivaneuh** [comment]
Ok thanks, waiting for it.

@rande what do you think of this PR?

**Comment by rande** [comment]
:-1: class might not exist yet as they need to be generated by easy extends.

**Comment by Soullivaneuh** [comment]
@rande What do you mean?

In this case, maybe the check could be done elsewhere in order to get a proper error message.

**Comment by rande** [comment]
The issue is not specific to the media bundle, but for all sonata's bundle and other bundles too. If the user fails to create the correct class nothing will work. I think we can check if doctrine generates a proper error message for missing class.

**Comment by rande** [comment]
closing for now. See related issue.

**Comment by pyrech** [comment]
If the user generates the bundle with easy extends, he doesn't need to specify the `class` parameter. Then, unless I'm wrong, if no class parameter is added to the conf, the default value will be given and the validation will not be run thus no error for the user, is that right?

**Comment by rande** [comment]
easy extends generated only the missing entities required to make the bundle works and so you can extends the model if you want. Now your code will throw an exception because the class will not exist.

**Pull request closed without merging**